### DISCUSSION TITLE
Question about compiling a script

### DISCUSSION THREAD

**Discussion by jojolom** [pyinstaller/pyinstaller#5685] [Help]
Hi, trying to build to one file with a word doc as a resource and plan on keeping the exe in the same file as the word doc. In the script I am accessing the word doc using **os.path.expanduser(dir_path + "/file.docx")** and can't seem to get it to run. Any help would be appreciated!

**Comment by rokm** [comment]
What are you using as your `dir_path`?

**Reply by jojolom** [reply]
my mistake, dir_path =os.path.dirname(os.path.realpath(__file__))

**Reply by jojolom** [reply]
I'm trying to get it to access the file in it's base folder

**Reply by rokm** [reply]
Using `__file__` as reference currently doesn't work, because it is always set to just the program basename in the frozen application (see #5626). But this is [documented behavior](https://pyinstaller.readthedocs.io/en/stable/runtime-information.html#using-file). So checking for `sys.frozen` and using `sys._MEIPASS` is necessary. 

However, the above applies to resources that are bundled *into* `onefile` application (`sys._MEIPASS` refers to temporary directory where program contents are unpacked). If I understand correctly, you are instead trying to put a .docx file next to the executable, and do not want to bundle it into the executable itself? In that case, you'll probably need to use `sys.executable`, which in a frozen application (if `sys.frozen` is `True` and/or `sys._MEIPASS` attribute exists) should point to the `onefile` executable itself.
